### ISSUE TITLE
Cigar StringIndexOutOfBoundsException error with VarDict 1.5.5

### ISSUE THREAD

**Issue by chapmanb** [AstraZeneca-NGS/VarDictJava#135]
Hello;
I've been working on testing out the latest VarDict 1.5.5 from master and am running into a cigar index error. Here is a fully contained reproducible example that demonstrates the issue (the data is publicly available, so feel free to incorporate into tests if that's helpful as well):

https://s3.amazonaws.com/chapmanb/testcases/vardict_stringindex.tar.gz

There is a `run.sh` script in there which errors out with:
```
Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 151
        at java.lang.String.charAt(String.java:658)
        at com.astrazeneca.vardict.modules.CigarUtils.modifyCigar(CigarUtils.java:197)
        at com.astrazeneca.vardict.modules.SAMFileParser.parseSAM(SAMFileParser.java:232)
        at com.astrazeneca.vardict.modules.ToVarsBuilder.toVars(ToVarsBuilder.java:66)
        at com.astrazeneca.vardict.VarDict.vardictNotParallel(VarDict.java:365)
        at com.astrazeneca.vardict.VarDict.nonAmpVardict(VarDict.java:312)
        at com.astrazeneca.vardict.VarDict.start(VarDict.java:72)
        at com.astrazeneca.vardict.Main.run(Main.java:152)
        at com.astrazeneca.vardict.Main.main(Main.java:31)
```
Thanks for all the work on 1.5.5 and looking forward to having the new release out.

**Comment by PolinaBevad** [comment]
@chapmanb, hello!
I prepared a fix in referenced PR #136 and added your test case there. It fixes this and 2 other possible places where such error can occur. 
Thank you very much for your help!

**Comment by chapmanb** [comment]
Polina -- thank you for the quick turnaround on this. I tested the fix and can confirm it's working great, the latest master branch is working on my test cases and producing great results in the current validations:

https://github.com/bcbio/bcbio_validations/tree/master/somatic-lowfreq#vardict-low-frequency-false-positive-removal

Thank you again
